### DISCUSSION TITLE
What about TCL tv with Android TV?

### DISCUSSION THREAD

**Discussion by gcalzo** [PredatH0r/ChanSort#282] [Ideas]
You literally _saved my life_ with my father's LG 49UN71006LB! :smiley:

Here in Italy DTT is changing too often and since I've learned to use the template I can quickly sort the channels!!!

Personally I prefer Android TV and currently I have a **TCL C71** one.
Even if TCL has a site for sorting channels (https://www.tcl.com/cedit/#/ and the UI is really similar to your pgm) I would like to suggest to add support for TCL format.
I can give you anytime a sample list db saved from my TV.

Thanks for you great program and I wish you a very Happy New Year!!!!

**Comment by PredatH0r** [comment]
Hi, thanks for your feedback.

Please send me your list, I'll take a look then. I can't make any promises about a time frame at the moment.
I can only hope it doesn't use the same proprietary, undecipherable file format the Panasonic Android TVs use nowadays.

**Reply by kutyux60** [reply]
Hi,

I would like to know if there is any progress on the editing of the TCL TV channel?

I have handpicked 60 favourite channels from nearly 600 Sat channels, sorted them in order and rewritten the channel number. It took many hours out of my life.

TV channel  database is SQLite. I exported it to xml format with DBtoFile, but I'm not an expert in databases.
Regards

**Comment by PredatH0r** [comment]
So far I only received a single TCL channel list in this format and with that little demand it's simply not worth the effort to put dozens of hours of work into it. Especially since there is a free alternative available "made" by TCL (which seems to be a web-based specialized clone of ChanSort).

**Comment by kutyux60** [comment]
Interesting, because it's quite a common type of TV.
The channel management is the worst I've come across.

The TCL web channel editor has been unavailable for several weeks.

Thank you very much.

**Comment by mcdermottrick** [comment]
Hi I'e tried with TCL TV TCL 65C635K Andorid TV. Each time i get error when re uploading the file via usb. Here is my original file and sorted using template.

[Channel_List_V8-R51MT02-LF1V334_202318sorted.tar.zip](https://github.com/PredatH0r/ChanSort/files/10387245/Channel_List_V8-R51MT02-LF1V334_202318sorted.tar.zip)


[Channel_List_V8-R51MT02-LF1V334_202318 Original.tar.zip](https://github.com/PredatH0r/ChanSort/files/10387188/Channel_List_V8-R51MT02-LF1V334_202318.Original.tar.zip)

really appreciate if you are able to help 
Thanks so much

**Comment by PredatH0r** [comment]
Thanks for providing your files.

On the issue tracker thread https://github.com/PredatH0r/ChanSort/issues/338 tomxhachijuroku mentioned that ChanSort works fine for his TCL 43P638, while matepor464 has the same issues that his 65C635 rejects the modified files (but also sometime unmodified files that were just copied around).

I will take a closer look at the files as soon as I find the time. From a first glance the format is the same as with the 43P638. It could be that the firmware behaves differently or maybe even is erratic. I haven't received feedback yet if resetting the TV, skipping/cancelling the channel search and loading the list on an "empty" TV makes any difference.

**Reply by mcdermottrick** [reply]
Hi I cleared the channel list and it still had same fault. 
if you need me to test anything let me know.

**Reply by PredatH0r** [reply]
A few others joined the discussion on https://github.com/PredatH0r/ChanSort/issues/338#issuecomment-1379547865.
I linked a new test version of ChanSort there which makes bare minimum changes to the list.
It would be helpful if someone could test a minimal change, like only swapping 2 programs and upload that again.

It's important to always start with a clean file, either freshly exported from the TV or using ChanSort's "Restore backup". If an edited file is rejected, any further edits of the same file will almost certainly rejected too.

**Reply by mcdermottrick** [reply]
Hi Tested and confirmed working perfectly  on TCL TCL 65C635K Android TV  ! i applied template from 28.2E and made a few adjustments also and it successfully copied across. Thanks so Much  for taking the time to make this work. 

Great Job ! 
Richard 
[Channel_List_V8-R51MT02-LF1V334_28.2E UK TCLTCL 65C635K Android TV.tar.zip](https://github.com/PredatH0r/ChanSort/files/10405223/Channel_List_V8-R51MT02-LF1V334_28.2E.UK.TCLTCL.65C635K.Android.TV.tar.zip)
File attached for anyone who wants to use / test

**Comment by gcalzo** [comment]
May I suggest to close this thread as "duplicate of 338" so to keep it easier to follow? :)

**Comment by PredatH0r** [comment]
Good idea and I just tried to do so, but there is no way to close a discussion thread. That only works with issue tracker treads.

**Comment by gcalzo** [comment]
...there _should_ be, somewhere, a way to close an "Idea"... like "Implemented", or "To be done", etc...
As the OP I checked but I can only create an _Issue_ from this thread...

**Comment by Sanguinarix** [comment]
Works almost fine with TCL C845, EXCEPT that the TV mixes up the channels by itself after 2-3 days. The sorted list imports correctly, and stays sorted for 2-3 days then the TV says "f--- you" and breaks the sorting. Auto channel update feature is disabled as far as I know, I'm quite  curious what might be causing this.

**Reply by PredatH0r** [reply]
I don't own a TCL TV, so I can't test this out personally. 
Are you using cable or antenna or satellite?
Is the TV restoring the original numbers when it reorders the list or does it seem to be completely random?

Some general thoughts:
Most TVs have built-in, presorted channel lists that are used to initialize the TV based on a country setting. If such a list is used and then reordered, I could be that the TV will change the list again when there is some sort of (firmware) update. Therefore I always recommend to run a neutral search, not selecting any specific "provider" - but that depends on the TV.
There is also the possibility that your cable TV provider sends out predefined channel numbers (LCN) and whenever they change anything the TV will reapply the numbers (in whole or part). Disabling channel auto-update should prevent this, but who knows...

It would be interesting to see a file exported right after a fresh channel search and in addition to it another export after manually moving a channel to a different number. Maybe there is some sort of flag to indicate that a channel has been moved by the user and that it should be locked in place. I could find that by comparing the two lists.

**Reply by Sanguinarix** [reply]
It seems random, but there likely is some logic, as with most computer stuff.
It's DVB-C (digital cable), the provider is UPC Slovakia.

The "Auto Update" feature strangely disappears as soon as a full scan is complete, I never explicitly enabled it or used it. I believe it's a full scan.

Here's the mysterious files:
[channel lists before after.zip](https://github.com/user-attachments/files/16846351/channel.lists.before.after.zip)

**Comment by smarius61-coder** [comment]
Hello,

I have a TCL 50C6K (2025 model, Google TV) connected to Digi cable TV in Romania.
I successfully scanned the channels using DVB-C and the TV found 211 digital channels and 48 analog channels. I also created a Favorites list directly on the TV and exported the channel list to a USB stick.
The TV created a file called **ChannelList_V8.tar**. Inside the TAR archive there is a database file, but I cannot directly inspect or edit it.
I would like to use ChanSort to organize the channels in the order I want, especially the Favorites list, and then import the modified channel list back into the TV.
My questions are:

1. Does ChanSort fully support the TCL 50C6K and its **ChannelList_V8.tar** format?
2. Can ChanSort read and preserve the **Favorites** list created on the TV?
3. Can I reorder the Favorite channels and assign them the channel numbers I want?
4. After editing, can I export the list in the correct format and import it back into the TCL 50C6K?

I initially downloaded **chansort-2025-03-08-installer.exe**, but the installer keeps launching another installer/download process and never actually installs the program. I understand that the portable ZIP version may be more appropriate, so I will try that version.

Thank you for your help.
[Channel_List_V8-R51MT02-LF1V334_28.2E.UK.TCLTCL.65C635K.Android.TV.tar.zip](https://github.com/user-attachments/files/32281560/Channel_List_V8-R51MT02-LF1V334_28.2E.UK.TCLTCL.65C635K.Android.TV.tar.zip)

**Comment by smarius61-coder** [comment]
**Correction to my previous message:**

I should clarify one important detail: I did **not** create the Favorites list directly on the TV.

I exported the channel list from my TCL 50C6K to a USB stick, opened the **ChannelList_V8.tar** file with ChanSort, and created the Favorites list **using ChanSort**.

What I would like to know is whether ChanSort can correctly save the Favorites arrangement and whether this modified list can then be imported back into the TCL 50C6K without losing the Favorites settings.

Sorry for the confusion in my previous message.

**Comment by PredatH0r** [comment]
Sorry for the late reply, I was on vacation and trying to catch up with things now.

The "chansort-2025-03-08-installer.exe" you downloaded must be some 3rd party repackaging of ChanSort that I have nothing to do with. I highly recommend downloading ChanSort only from GitHub (or via chansort.com), as there is no way of knowing what else is inside that installer-exe. It could be anything from AdWare to malware.

About your questions:
1) Short answer: No. I don't have any information from TCL about their file format and rely on analyze files that were sent to me by users. I implement support for these files to the best of my knowledge, but there is no way of knowing that these files cover all possible scenarios. (more about this below)
2) ChanSort can read and edit the favorite list inside the .tar (DtvData.db)
3) No. This is a limitation of the TV's firmware, not of ChanSort. The TV only "flags" channels if they are in a certain favorite list or not and doesn't manage any numbering or order for the favorites. It's based on the main channel number.
4) ChanSort saves files in the exact same format as it was read. It can't convert formats. (Just make sure you use the "Save" function and not the "Export reference list", which creates a TV-independent text file that is only useful for reordering another TV-file)

The file you added doesn't quite seem to match what you describe above.
When I open it in ChanSort, I see 454 UK satellite channels, not a Romanian cable TV list.
When I look inside the DtvData.db file (with a SQLite browser tool like SQLiteStudio) I see that after these 454 channels with assigned numbers, there's another ~420 channels without a number.
